### PR TITLE
Added support for comments on other sites than SO, added placeholders, fixed some bugs

### PR DESCRIPTION
**Pull request by fastnlight0**
Added support for comments and Answer related NAA options for sites other than Stack Overflow.

In addition, similar to AutoReviewComments, I added 3 placeholders:
|Placeholder|Description|Example|
|------------|-----------|---------|
|%SITENAME%|The site name|Stack Overflow, Super User, Physics Stack Exchange|
|%SITEURL%|The domain of the current site|stackoverflow.com, money.stackexchange.com|
|%OP%|The author of the question/answer|fastnlight|

I've also fixed a bug where if a user was recognized in a collective, AdvancedFlagging would believe their answer was actually a question.

### CODE REVIEW

File: src/popover.ts [hunk 1]
```diff
@@ -348,7 +348,16 @@ function getCommentText(
...
-    const comment = comments?.[commentType] || comments?.low;
+    let comment = comments?.[commentType] || comments?.low;
+
+    if (comment){
+        const sitename = StackExchange.options.site.name || '';
+        const siteurl = window.location.hostname;
+
+        comment = comment.replace(/%SITENAME%/, sitename);
+        comment = comment.replace(/%SITEURL%/, siteurl);
+        comment = comment.replace(/%OP%/, opName);
```
Comment (double-beep) [inline src/popover.ts:359]: This will only replace the first occurrence of each placeholder.

**Review by double-beep — CHANGES_REQUESTED** [review]
Thanks for the PR! I will test the changes in the weekend.

**Comment by double-beep** [comment]
Hmm... I'm reluctant to approve this. IIRC, the reason that adding answer-related comments via the AF menu on non-SO sites was disabled is because of [this Charcoal policy](https://charcoal-se.org/training/reports.html#comments-on-reported-posts):

> You should not leave comments under a post reported by SmokeDetector, unless you are otherwise familiar with that Stack Exchange site. In general, that means you participate on the site for more than responding to SmokeDetector reports and/or viewing a few Hot Network Questions.

**Comment by double-beep** [comment]
I'm sorry for the delay in responding. Since I've rewritten some parts of the code this PR modifies, could you open a new PR with your changes? Thanks!

**Comment by fastnlight0** [comment]
@double-beep Done! #292

**Pull request closed without merging**